src/styles/main.scss

```diff
@@ -0,0 +1,51 @@
+$days: 31;
+$month-length: 28, 29, 30, 31;
+$start-day: mon, thu, wed, thu, fri, sat, sun;
+$gap: 1px;
+$cell-size: 100px;
+
+body {
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  min-height: 100vh;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+@for $i from 1 through $days {
+  .calendar__day:nth-child(#{$i})::before {
+    content: '#{$i}';
+  }
+}
+
+.calendar {
+  gap: $gap;
+  display: flex;
+  flex-wrap: wrap;
+  width: 7 * ($cell-size + $gap) - $gap;
+
+  &__day {
+    margin: 0;
+    box-sizing: border-box;
+    height: $cell-size;
+    width: $cell-size;
+    background-color: #eee;
+    border: 1px solid #000;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+
+    &:hover {
+      transition-duration: 0.5s;
```

Comment:
The `transition-duration` should be specified within the `.calendar__day` block, not within the `:hover` pseudo-class. This way, the transition effect will be applied both when the mouse enters and leaves the element, making the animation smoother.